### ISSUE TITLE
no GUI shown if executed

### ISSUE THREAD

**Issue by Jakob-S** [linux-man/unreel#4]
Cause I had the following issue installing Unreel via AUR:
`⚠️  Packaging using asar archive is disabled — it is strongly not recommended.
Please enable asar and use asarUnpack to unpack files that must be externally available.
Packaging for linux x64 using electron 1.8.2 to dist/linux-unpacked
TypeError [ERR_INVALID_ARG_TYPE]: The "options" argument must be one of type string or Object. Received type number
    at getOptions (fs.js:87:11)
    at new WriteStream (fs.js:2162:24)
    at Object.fs.createWriteStream (fs.js:2153:10)
    at fcopy_streams (/tmp/yaourt-tmp-jakob/aur-unreel/src/unreel-1.2.0/node_modules/fcopy-pre-bundled/lib/copiers/fallback.js:7:27)
    at fcopy (/tmp/yaourt-tmp-jakob/aur-unreel/src/unreel-1.2.0/node_modules/fcopy-pre-bundled/lib/fcopy.js:26:13)
    at /tmp/yaourt-tmp-jakob/aur-unreel/src/unreel-1.2.0/node_modules/electron-builder-util/src/fs.ts:156:5
From previous event:
    at copyOrLinkFile (/tmp/yaourt-tmp-jakob/aur-unreel/src/unreel-1.2.0/node_modules/electron-builder-util/src/fs.ts:155:3)
    at /tmp/yaourt-tmp-jakob/aur-unreel/src/unreel-1.2.0/node_modules/electron-builder-util/src/fs.ts:181:13
    at Generator.next (<anonymous>)
From previous event:
    at FileCopier.copy (/tmp/yaourt-tmp-jakob/aur-unreel/src/unreel-1.2.0/node_modules/electron-builder-util/out/fs.js:243:11)
    at /tmp/yaourt-tmp-jakob/aur-unreel/src/unreel-1.2.0/node_modules/electron-builder-util/src/fs.ts:225:24
    at Generator.next (<anonymous>)
    at xfs.mkdir.er (/tmp/yaourt-tmp-jakob/aur-unreel/src/unreel-1.2.0/node_modules/fs-extra-p/node_modules/fs-extra/lib/mkdirs/mkdirs.js:37:14)
    at FSReqWrap.oncomplete (fs.js:149:20)
From previous event:
    at walk (/tmp/yaourt-tmp-jakob/aur-unreel/src/unreel-1.2.0/node_modules/electron-builder-util/out/fs.js:276:26)
    at /tmp/yaourt-tmp-jakob/aur-unreel/src/unreel-1.2.0/node_modules/electron-builder-util/src/fs.ts:70:60
    at /tmp/yaourt-tmp-jakob/aur-unreel/src/unreel-1.2.0/node_modules/graceful-fs/polyfills.js:287:18
    at FSReqWrap.oncomplete (fs.js:167:5)
From previous event:
    at /tmp/yaourt-tmp-jakob/aur-unreel/src/unreel-1.2.0/node_modules/electron-builder-util/src/fs.ts:65:10
    at runCallback (timers.js:756:18)
    at tryOnImmediate (timers.js:717:5)
    at processImmediate [as _immediateCallback] (timers.js:697:5)
From previous event:
    at /tmp/yaourt-tmp-jakob/aur-unreel/src/unreel-1.2.0/node_modules/electron-builder-util/src/fs.ts:62:51
    at Generator.next (<anonymous>)
    at go$readdir$cb (/tmp/yaourt-tmp-jakob/aur-unreel/src/unreel-1.2.0/node_modules/graceful-fs/graceful-fs.js:149:14)
    at FSReqWrap.oncomplete (fs.js:149:20)
From previous event:
    at walk (/tmp/yaourt-tmp-jakob/aur-unreel/src/unreel-1.2.0/node_modules/electron-builder-util/out/fs.js:110:22)
    at copyDir (/tmp/yaourt-tmp-jakob/aur-unreel/src/unreel-1.2.0/node_modules/electron-builder-util/src/fs.ts:213:3)
    at /tmp/yaourt-tmp-jakob/aur-unreel/src/unreel-1.2.0/node_modules/electron-builder/src/platformPackager.ts:197:17
    at Generator.next (<anonymous>)
    at runCallback (timers.js:756:18)
    at tryOnImmediate (timers.js:717:5)
    at processImmediate [as _immediateCallback] (timers.js:697:5)
From previous event:
    at LinuxPackager.doPack (/tmp/yaourt-tmp-jakob/aur-unreel/src/unreel-1.2.0/node_modules/electron-builder/out/platformPackager.js:280:11)
    at /tmp/yaourt-tmp-jakob/aur-unreel/src/unreel-1.2.0/node_modules/electron-builder/src/platformPackager.ts:104:16
    at Generator.next (<anonymous>)
From previous event:
    at LinuxPackager.pack (/tmp/yaourt-tmp-jakob/aur-unreel/src/unreel-1.2.0/node_modules/electron-builder/out/platformPackager.js:177:11)
    at /tmp/yaourt-tmp-jakob/aur-unreel/src/unreel-1.2.0/node_modules/electron-builder/src/packager.ts:270:24
From previous event:
    at Packager.doBuild (/tmp/yaourt-tmp-jakob/aur-unreel/src/unreel-1.2.0/node_modules/electron-builder/out/packager.js:388:11)
    at /tmp/yaourt-tmp-jakob/aur-unreel/src/unreel-1.2.0/node_modules/electron-builder/src/packager.ts:183:52
    at Generator.next (<anonymous>)
    at runCallback (timers.js:756:18)
    at tryOnImmediate (timers.js:717:5)
    at processImmediate [as _immediateCallback] (timers.js:697:5)
From previous event:
    at Packager.build (/tmp/yaourt-tmp-jakob/aur-unreel/src/unreel-1.2.0/node_modules/electron-builder/out/packager.js:283:11)
    at /tmp/yaourt-tmp-jakob/aur-unreel/src/unreel-1.2.0/node_modules/electron-builder/src/builder.ts:199:33
    at Generator.next (<anonymous>)
From previous event:
    at build (/tmp/yaourt-tmp-jakob/aur-unreel/src/unreel-1.2.0/node_modules/electron-builder/out/builder.js:70:21)
    at Object.<anonymous> (/tmp/yaourt-tmp-jakob/aur-unreel/src/unreel-1.2.0/node_modules/electron-builder/out/cli/build-cli.js:71:41)
    at Module._compile (module.js:660:30)
    at Object.Module._extensions..js (module.js:671:10)
    at Module.load (module.js:573:32)
    at tryModuleLoad (module.js:513:12)
    at Function.Module._load (module.js:505:3)
    at Function.Module.runMain (module.js:701:10)
    at startup (bootstrap_node.js:190:16)
    at bootstrap_node.js:662:3
==> FEHLER: Ein Fehler geschah in build().
    Breche ab...
==> FEHLER:Makepkg konnte unreel nicht erstellen.
`
I tried to install it from source. Started the main.js with electron but it crashed cause of missing libraries "tmp" and "fs-extra" which i installed with `npm install tmp fs-extra`.
But if I open it now with `electron main.js` I see the GUI only 1 second.
The only message I get from Electron is the following: `(electron:27197): Gtk-WARNING **: Theme parsing error: gtk.css:3601:22: 'none' is not a valid color name`
Dunno if it's only the 'none' which raises this issue.

**Comment by Photon89** [comment]
The AUR package is currently broken, sorry for that. I couldn't find a fix quickly, will have to look more thoroughly into it. I fear, I will only have time in one week for it...

**Comment by Jakob-S** [comment]
Thanks, would be great, cause I'm going to give a lecture in a few weeks about complex numbers, and Reveal.JS really cool to use for it cause of the Chalkboard Plugin. I'll try in 1-2 weeks a second time.
Good Luck :+1:

**Comment by linux-man** [comment]
You could try AppImage for now.
Also - to whom may concern - Chalkboard plugin (and a lot of other plugins) is not included in Unreel. To use it you can export the presentation as HTML and then manually add it.

**Comment by Jakob-S** [comment]
Hello linux-man,
What exactly is AppImage?
Could you share a link or sth like this?
Thanks

**Comment by linux-man** [comment]
I was referring the AppImage format you can find at https://github.com/linux-man/unreel/releases.

**Comment by Jakob-S** [comment]
I've unpacked the AppImage using mount and cp.
Then I've copied the official plugins from GitHub (reveal.js-plugins) to unreel/usr/bin/resources/app/reveal.js/plugin/ and packed it as AppImage.
But it seems not to be recognized by the main program. Which files do I also have to edit, that the added plugins gonna work?

**Comment by linux-man** [comment]
Unfortunately, it's not that easy. I added Chalkboard to Unreel but found a problem I couldn't solve. The plugin forget the drawings when moving between slides. This is probably related to the security restrictions of the electron webview element used to render reveal presentations. It works as a sandboxed browser so there must be something that this plugin is not allowed to do.
For now, the best solution (and it's a one way path because you can't reopen presentations exported as html although you can always save to reel format to edit later) is to
- Create your presentation.
- Save as HTML.
- Inside the presentation folder, copy Chalkboard plugin to 'plugin' folder.
- Open index.html. At the end you will find a 'Reveal.initialize' script. Change
```
            dependencies: [
            {src: "plugin/notes/notes.js", async: true},
            {src: "plugin/highlight/highlight.js", async: true, callback: function() {hljs.initHighlightingOnLoad();}},
            {src: "plugin/zoom-js/zoom.js", async: true},
            {src: "plugin/search/search.js", async: true},
            {src: "plugin/katex/katex.js", async: true},
            {src: "plugin/chart/Chart.min.js"},
            {src: "plugin/chart/csv2chart.js"}]
```
to
```
            dependencies: [
            {src: "plugin/notes/notes.js", async: true},
            {src: "plugin/highlight/highlight.js", async: true, callback: function() {hljs.initHighlightingOnLoad();}},
            {src: "plugin/zoom-js/zoom.js", async: true},
            {src: "plugin/search/search.js", async: true},
            {src: "plugin/katex/katex.js", async: true},
            {src: "plugin/chart/Chart.min.js"},
            {src: "plugin/chart/csv2chart.js"},
            {src: "plugin/chalkboard/chalkboard.js"}],
            keyboard: {
              67: function() { RevealChalkboard.toggleNotesCanvas() },	// toggle notes canvas when 'c' is pressed
              66: function() { RevealChalkboard.toggleChalkboard() },	// toggle chalkboard when 'b' is pressed
              46: function() { RevealChalkboard.clear() },	// clear chalkboard when 'DEL' is pressed
               8: function() { RevealChalkboard.reset() },	// reset chalkboard data on current slide when 'BACKSPACE' is pressed
              68: function() { RevealChalkboard.download() },	// download recorded chalkboard drawing when 'd' is pressed
            }
```
- Use a browser to show your presentation.

Sorry. That's all I can offer for now. :disappointed:

**Comment by Jakob-S** [comment]
Hey linux-man,
thanks for the detailed advice :)
Tried to copy and paste you code into my index.html.
If i press "b", the chalkboard appears, but i can't draw on it.
Is it maybe the browser which doesn't work properly with it?
Using Vivaldi 1.14.1077.50  (stable)

**Comment by linux-man** [comment]
I can use it with firefox and chrome but with issues.
I tried the Chalkboard online demo and found the same problems: 
Sometimes you can't draw. Sometimes "c" don't work. If you return to a slide with drawings you can't draw over.
Most of the times you need to "BACKSPACE" to draw again.

**Comment by Photon89** [comment]
@linux-man: For some reason building fails even with the old 1.2.0 release though neither the Unreel code nor the PKGBUILD have changed. Looks like some make dependencies (electron?) were updated and now the building fails. Do you have any idea what might have happened? Unfortunately, I have no experience in electron packages and the unreel package is the only one I (try to) maintain, I used a different package as a template... Are there any build instructions which could help updating the package?

**Comment by linux-man** [comment]
You're probably building with the last Electron (1.8.2).
Try to build 1.3.1 release. Electron 1.8 breaks something so I'm sticking with 1.7 branch.

**Comment by Photon89** [comment]
Thanks for the quick reply! :) I'm getting the following error with 1.3.1:
```

TypeError [ERR_INVALID_ARG_TYPE]: The "options" argument must be one of type string or Object. Received type number
    at getOptions (fs.js:87:11)
    at new WriteStream (fs.js:2162:24)
    at Object.fs.createWriteStream (fs.js:2153:10)
    at fcopy_streams (/media/Daten/PKGBUILDS/unreel/src/unreel-1.3.1/node_modules/fcopy-pre-bundled/lib/copiers/fallback.js:7:27)
    at fcopy (/media/Daten/PKGBUILDS/unreel/src/unreel-1.3.1/node_modules/fcopy-pre-bundled/lib/fcopy.js:26:13)
    at /media/Daten/PKGBUILDS/unreel/src/unreel-1.3.1/node_modules/electron-builder-util/src/fs.ts:156:5
From previous event:
    at copyOrLinkFile (/media/Daten/PKGBUILDS/unreel/src/unreel-1.3.1/node_modules/electron-builder-util/src/fs.ts:155:3)
    at /media/Daten/PKGBUILDS/unreel/src/unreel-1.3.1/node_modules/electron-builder-util/src/fs.ts:181:13
    at Generator.next (<anonymous>)
From previous event:
    at FileCopier.copy (/media/Daten/PKGBUILDS/unreel/src/unreel-1.3.1/node_modules/electron-builder-util/out/fs.js:243:11)
    at /media/Daten/PKGBUILDS/unreel/src/unreel-1.3.1/node_modules/electron-builder-util/src/fs.ts:225:24
    at Generator.next (<anonymous>)
    at xfs.mkdir.er (/media/Daten/PKGBUILDS/unreel/src/unreel-1.3.1/node_modules/fs-extra-p/node_modules/fs-extra/lib/mkdirs/mkdirs.js:37:14)
    at FSReqWrap.oncomplete (fs.js:149:20)
From previous event:
    at walk (/media/Daten/PKGBUILDS/unreel/src/unreel-1.3.1/node_modules/electron-builder-util/out/fs.js:276:26)
    at /media/Daten/PKGBUILDS/unreel/src/unreel-1.3.1/node_modules/electron-builder-util/src/fs.ts:70:60
    at /media/Daten/PKGBUILDS/unreel/src/unreel-1.3.1/node_modules/graceful-fs/polyfills.js:287:18
    at FSReqWrap.oncomplete (fs.js:167:5)
From previous event:
    at /media/Daten/PKGBUILDS/unreel/src/unreel-1.3.1/node_modules/electron-builder-util/src/fs.ts:65:10
    at runCallback (timers.js:756:18)
    at tryOnImmediate (timers.js:717:5)
    at processImmediate [as _immediateCallback] (timers.js:697:5)
From previous event:
    at /media/Daten/PKGBUILDS/unreel/src/unreel-1.3.1/node_modules/electron-builder-util/src/fs.ts:62:51
    at Generator.next (<anonymous>)
    at go$readdir$cb (/media/Daten/PKGBUILDS/unreel/src/unreel-1.3.1/node_modules/graceful-fs/graceful-fs.js:149:14)
    at FSReqWrap.oncomplete (fs.js:149:20)
From previous event:
    at walk (/media/Daten/PKGBUILDS/unreel/src/unreel-1.3.1/node_modules/electron-builder-util/out/fs.js:110:22)
    at copyDir (/media/Daten/PKGBUILDS/unreel/src/unreel-1.3.1/node_modules/electron-builder-util/src/fs.ts:213:3)
    at /media/Daten/PKGBUILDS/unreel/src/unreel-1.3.1/node_modules/electron-builder/src/platformPackager.ts:197:17
    at Generator.next (<anonymous>)
    at runCallback (timers.js:756:18)
    at tryOnImmediate (timers.js:717:5)
    at processImmediate [as _immediateCallback] (timers.js:697:5)
From previous event:
    at LinuxPackager.doPack (/media/Daten/PKGBUILDS/unreel/src/unreel-1.3.1/node_modules/electron-builder/out/platformPackager.js:280:11)
    at /media/Daten/PKGBUILDS/unreel/src/unreel-1.3.1/node_modules/electron-builder/src/platformPackager.ts:104:16
    at Generator.next (<anonymous>)
From previous event:
    at LinuxPackager.pack (/media/Daten/PKGBUILDS/unreel/src/unreel-1.3.1/node_modules/electron-builder/out/platformPackager.js:177:11)
    at /media/Daten/PKGBUILDS/unreel/src/unreel-1.3.1/node_modules/electron-builder/src/packager.ts:270:24
From previous event:
    at Packager.doBuild (/media/Daten/PKGBUILDS/unreel/src/unreel-1.3.1/node_modules/electron-builder/out/packager.js:388:11)
    at /media/Daten/PKGBUILDS/unreel/src/unreel-1.3.1/node_modules/electron-builder/src/packager.ts:183:52
    at Generator.next (<anonymous>)
    at runCallback (timers.js:756:18)
    at tryOnImmediate (timers.js:717:5)
    at processImmediate [as _immediateCallback] (timers.js:697:5)
From previous event:
    at Packager.build (/media/Daten/PKGBUILDS/unreel/src/unreel-1.3.1/node_modules/electron-builder/out/packager.js:283:11)
    at /media/Daten/PKGBUILDS/unreel/src/unreel-1.3.1/node_modules/electron-builder/src/builder.ts:199:33
    at Generator.next (<anonymous>)
From previous event:
    at build (/media/Daten/PKGBUILDS/unreel/src/unreel-1.3.1/node_modules/electron-builder/out/builder.js:70:21)
    at Object.<anonymous> (/media/Daten/PKGBUILDS/unreel/src/unreel-1.3.1/node_modules/electron-builder/out/cli/build-cli.js:71:41)
    at Module._compile (module.js:660:30)
    at Object.Module._extensions..js (module.js:671:10)
    at Module.load (module.js:573:32)
    at tryModuleLoad (module.js:513:12)
    at Function.Module._load (module.js:505:3)
    at Function.Module.runMain (module.js:701:10)
    at startup (bootstrap_node.js:190:16)
    at bootstrap_node.js:662:3
```

**Comment by linux-man** [comment]
It's the same error @Jakob-S found. Please confirm that you are using Electron 1.7.12. Remove any package-lock.json and node_modules folder.
If that don't work I'm afraid I'm lost.

**Comment by Photon89** [comment]
I cleaned the build dir and tried rebuilding. It says, it uses electron 1.7.12 but I don't know if it really does. Through package management only 1.8.2 is available... This is the output:

```
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: 7zip-bin-win@2.1.1 (node_modules/7zip-bin-win):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for 7zip-bin-win@2.1.1: wanted {"os":"win32","arch":"any"} (current: {"os":"linux","arch":"x64"})
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: 7zip-bin-mac@1.0.1 (node_modules/7zip-bin-mac):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for 7zip-bin-mac@1.0.1: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"x64"})

up to date in 3.551s
electron-builder 18.8.1
No native production dependencies
⚠️  Packaging using asar archive is disabled — it is strongly not recommended.
Please enable asar and use asarUnpack to unpack files that must be externally available.
Packaging for linux x64 using electron 1.7.12 to dist/linux-unpacked
TypeError [ERR_INVALID_ARG_TYPE]: The "options" argument must be one of type string or Object. Received type number
    at getOptions (fs.js:87:11)
    at new WriteStream (fs.js:2162:24)
    at Object.fs.createWriteStream (fs.js:2153:10)
    at fcopy_streams (/media/Daten/PKGBUILDS/unreel/src/unreel-1.3.1/node_modules/fcopy-pre-bundled/lib/copiers/fallback.js:7:27)
    at fcopy (/media/Daten/PKGBUILDS/unreel/src/unreel-1.3.1/node_modules/fcopy-pre-bundled/lib/fcopy.js:26:13)
    at /media/Daten/PKGBUILDS/unreel/src/unreel-1.3.1/node_modules/electron-builder-util/src/fs.ts:156:5
From previous event:
    at copyOrLinkFile (/media/Daten/PKGBUILDS/unreel/src/unreel-1.3.1/node_modules/electron-builder-util/src/fs.ts:155:3)
    at /media/Daten/PKGBUILDS/unreel/src/unreel-1.3.1/node_modules/electron-builder-util/src/fs.ts:181:13
    at Generator.next (<anonymous>)
From previous event:
    at FileCopier.copy (/media/Daten/PKGBUILDS/unreel/src/unreel-1.3.1/node_modules/electron-builder-util/out/fs.js:243:11)
    at /media/Daten/PKGBUILDS/unreel/src/unreel-1.3.1/node_modules/electron-builder-util/src/fs.ts:225:24
    at Generator.next (<anonymous>)
    at xfs.mkdir.er (/media/Daten/PKGBUILDS/unreel/src/unreel-1.3.1/node_modules/fs-extra-p/node_modules/fs-extra/lib/mkdirs/mkdirs.js:37:14)
    at FSReqWrap.oncomplete (fs.js:149:20)
From previous event:
    at walk (/media/Daten/PKGBUILDS/unreel/src/unreel-1.3.1/node_modules/electron-builder-util/out/fs.js:276:26)
    at /media/Daten/PKGBUILDS/unreel/src/unreel-1.3.1/node_modules/electron-builder-util/src/fs.ts:70:60
    at /media/Daten/PKGBUILDS/unreel/src/unreel-1.3.1/node_modules/graceful-fs/polyfills.js:287:18
    at FSReqWrap.oncomplete (fs.js:167:5)
From previous event:
    at /media/Daten/PKGBUILDS/unreel/src/unreel-1.3.1/node_modules/electron-builder-util/src/fs.ts:65:10
    at runCallback (timers.js:756:18)
    at tryOnImmediate (timers.js:717:5)
    at processImmediate [as _immediateCallback] (timers.js:697:5)
From previous event:
    at /media/Daten/PKGBUILDS/unreel/src/unreel-1.3.1/node_modules/electron-builder-util/src/fs.ts:62:51
    at Generator.next (<anonymous>)
    at go$readdir$cb (/media/Daten/PKGBUILDS/unreel/src/unreel-1.3.1/node_modules/graceful-fs/graceful-fs.js:149:14)
    at FSReqWrap.oncomplete (fs.js:149:20)
From previous event:
    at walk (/media/Daten/PKGBUILDS/unreel/src/unreel-1.3.1/node_modules/electron-builder-util/out/fs.js:110:22)
    at copyDir (/media/Daten/PKGBUILDS/unreel/src/unreel-1.3.1/node_modules/electron-builder-util/src/fs.ts:213:3)
    at /media/Daten/PKGBUILDS/unreel/src/unreel-1.3.1/node_modules/electron-builder/src/platformPackager.ts:197:17
    at Generator.next (<anonymous>)
    at runCallback (timers.js:756:18)
    at tryOnImmediate (timers.js:717:5)
    at processImmediate [as _immediateCallback] (timers.js:697:5)
From previous event:
    at LinuxPackager.doPack (/media/Daten/PKGBUILDS/unreel/src/unreel-1.3.1/node_modules/electron-builder/out/platformPackager.js:280:11)
    at /media/Daten/PKGBUILDS/unreel/src/unreel-1.3.1/node_modules/electron-builder/src/platformPackager.ts:104:16
    at Generator.next (<anonymous>)
From previous event:
    at LinuxPackager.pack (/media/Daten/PKGBUILDS/unreel/src/unreel-1.3.1/node_modules/electron-builder/out/platformPackager.js:177:11)
    at /media/Daten/PKGBUILDS/unreel/src/unreel-1.3.1/node_modules/electron-builder/src/packager.ts:270:24
From previous event:
    at Packager.doBuild (/media/Daten/PKGBUILDS/unreel/src/unreel-1.3.1/node_modules/electron-builder/out/packager.js:388:11)
    at /media/Daten/PKGBUILDS/unreel/src/unreel-1.3.1/node_modules/electron-builder/src/packager.ts:183:52
    at Generator.next (<anonymous>)
    at runCallback (timers.js:756:18)
    at tryOnImmediate (timers.js:717:5)
    at processImmediate [as _immediateCallback] (timers.js:697:5)
From previous event:
    at Packager.build (/media/Daten/PKGBUILDS/unreel/src/unreel-1.3.1/node_modules/electron-builder/out/packager.js:283:11)
    at /media/Daten/PKGBUILDS/unreel/src/unreel-1.3.1/node_modules/electron-builder/src/builder.ts:199:33
    at Generator.next (<anonymous>)
From previous event:
    at build (/media/Daten/PKGBUILDS/unreel/src/unreel-1.3.1/node_modules/electron-builder/out/builder.js:70:21)
    at Object.<anonymous> (/media/Daten/PKGBUILDS/unreel/src/unreel-1.3.1/node_modules/electron-builder/out/cli/build-cli.js:71:41)
    at Module._compile (module.js:660:30)
    at Object.Module._extensions..js (module.js:671:10)
    at Module.load (module.js:573:32)
    at tryModuleLoad (module.js:513:12)
    at Function.Module._load (module.js:505:3)
    at Function.Module.runMain (module.js:701:10)
    at startup (bootstrap_node.js:190:16)
    at bootstrap_node.js:662:3


   ╭─────────────────────────────────────────────────────╮
   │                                                     │
   │          Update available 18.8.1 → 20.2.0           │
   │   Run npm i electron-builder --save-dev to update   │
   │                                                     │
   ╰─────────────────────────────────────────────────────╯


```

**Comment by linux-man** [comment]
Found one bug report with a similar error https://github.com/EragonJ/Kaku/issues/529.
I can build with the latest electron-builder but it looks like you can't.
Maybe downgrade?
Could you change devDependencies to "electron-builder": "~18.0.1" and try again?
Other option is to use the binary zip or deb.

**Comment by Photon89** [comment]
Sorry for the late reply. The building works with the change you proposed. What does it mean, what are possible next steps to fix the package?

@Jakob-S: If it's not yet too late, you can retry using the AUR package, it should work for now (I changed the electron-builder version as advised by @linux-man and it builds now).

**Comment by linux-man** [comment]
There are really 2 issues here.
In the near future I must find why Unreel can't build on the last Electron branch - 1.8.x - and fix it.
The other issue is with electron-builder and that is more tricky. It works on Ubuntu/Mint but breaks on Arch. It's hard to believe that one of the most updated distros has problems with an updated package.
I'll have to install Arch but hope that @Photon89 will find the problem - I have no experience with PGKBUILD.
Freezing electron and electron-builder versions will do the trick for now.

**Comment by Photon89** [comment]
@linux-man: PKGBUILDs are just recipes for building an app plus some metadata like dependencies, conflicts etc. In the simplest case the interesting part of a PKGBUILD is just a "./configure && make && make install". So if I would understand why the compilation fails, fixing the PKGBUILD is not a problem at all. But since I have absolutely no experience with electron, I'm really lost and don't understand the root of the problem so far...

**Comment by Photon89** [comment]
This seems to be fixed in 1.4.0.

**Comment by linux-man** [comment]
Great. Closing.
